wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java

```diff
@@ -196,58 +194,55 @@ public boolean writeHPCCFileData(byte []  data, String fileName, String targetLa
+
+        while (bytesleft > 0)
         {
-            WriteFileDataRequest writefileparams = new WriteFileDataRequest();
-            writefileparams.setAppend(append);
-            writefileparams.setDestDropZone(targetLandingZone);
-            writefileparams.setDestRelativePath(fileName);
-            writefileparams.setOffset(offset);
-
-            int dataindex = 0;
-            int limit = uploadchunksize <= 0 ? defaultUploadChunkSize : uploadchunksize;
-            long payloadsize = 0;
-            long bytesleft = data.length;
-            byte [] subdata = null;
-
-            while (bytesleft > 0)
-            {
-                payloadsize = bytesleft >= limit ? limit : bytesleft;
-                log.trace("Writing offset: "+dataindex+"\t size: "+((int)dataindex + (int)payloadsize));
+            payloadsize = bytesleft >= limit ? limit : bytesleft;
+            log.trace("Writing offset: "+dataindex+"\t size: "+((int)dataindex + (int)payloadsize));
 
-                subdata = new byte [(int)payloadsize];
-                for (int i = 0; i < payloadsize ; i++, dataindex++)
-                    subdata[i] = data[dataindex];
+            subdata = new byte [(int)payloadsize];
+            for (int i = 0; i < payloadsize ; i++, dataindex++)
+                subdata[i] = data[dataindex];
 
-                writefileparams.setData(subdata);
-                writefileparams.setAppend(dataindex > 0);
-                writefileparams.setOffset((long)dataindex);
+            writefileparams.setData(subdata);
+            writefileparams.setAppend(dataindex > 0);
+            writefileparams.setOffset((long)dataindex);
 
-                try
+            try
+            {
+                WriteFileDataResponse writeFileDataResponse = wsFileIOServiceSoapProxy.writeFileData(writefileparams);
+                String result = writeFileDataResponse.getResult();
+
+                log.debug(result);
+                //Wish there was a better way
+                if (!result.startsWith("Failed"))
                 {
-                    WriteFileDataResponse writeFileDataResponse = wsFileIOServiceSoapProxy.writeFileData(writefileparams);
-                    String result = writeFileDataResponse.getResult();
-
-                    log.debug(result);
-                    //Wish there was a better way
-                    if (!result.startsWith("Failed"))
-                    {
-                        success = true;
-                        bytesleft = bytesleft - subdata.length;
-                    }
-                    else
-                    {
-                        bytesleft = 0;
-                    }
+                    success = true;
+                    bytesleft = bytesleft - subdata.length;
                 }
-                catch (Exception e)
+                else
                 {
-                    log.error(e.getLocalizedMessage());
-                    return false;
+                    bytesleft = 0;
```

Comment:
Should there be a log message or an exception thrown here?

Reply:
I think the else condition implies success.Anyhow this logic didn't change, it was indented due to removal of if/else